customer-churn/README.md

```diff
@@ -0,0 +1,247 @@
+![cloudkubeflowstack](_assets/wholekubeflowstack.gif)
+
+## Continuous model deployment with Seldon Core
+
+While building the real-world workflow for predicting whether a customer will churn or not, you might not want to train the model once and deploy it to production. Instead, you might want to train the model and deploy it to production when something gets triggered. This is where one of our recent integration comes in, [Seldon Core](https://github.com/zenml-io/zenml/tree/main/examples/seldon_deployment).
+
+[Seldon Core](https://github.com/SeldonIO/seldon-core) is a production-grade open-source model serving platform. It packs a wide range of features built around deploying models to REST/GRPC microservices, including monitoring and logging, model explainers, outlier detectors, and various continuous deployment strategies such as A/B testing and canary deployments, and more.
+
+It also comes equipped with a set of built-in model server implementations designed to work with standard formats for packaging ML models that greatly simplify serving models for real-time inference.
+
+In this project, we build a continuous deployment pipeline that trains a model and then serves it with Seldon Core as the industry-ready model deployment tool of choice. If you are interested in learning more about Seldon Core, you can check out the [ZenML Example](https://github.com/zenml-io/zenml/tree/main/examples/seldon_deployment). Following diagram shows the flow of the whole pipeline:
+![seldondeployment](_assets/seldoncorecontinousdeployment.gif)
+
+Now, Let's start setting up our Full AWS stack to run the pipeline using Seldon Core.
+
+1. Install the Seldon Core Integration, a set of ZenML extensions that integrate with Seldon Core.
+
+```bash
+zenml integration install Seldon
+```
+
+2. Register the stack components
+
+```bash
+aws eks --region us-east-1 update-kubeconfig --name zenml-cluster --alias zenml-eks
+```
+
+To configure ECR registry access locally, run, e.g.:
+
+```bash
+aws ecr get-login-password --region us-east-1 | docker login --username AWS \
+  --password-stdin 715803424590.dkr.ecr.us-east-1.amazonaws.com
+```
+
+Extract the URL where the Seldon Core model server exposes its prediction API, e.g.:
+
+```bash
+export INGRESS_HOST=$(kubectl -n istio-system get service istio-ingressgateway \
+  -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+```
+
+Configuring the stack can be done like this:
+
+```shell
+zenml integration install s3 aws kubeflow Seldon
+
+zenml artifact-store register aws --type=s3 --path=s3://mybucket
+zenml model-deployer register seldon_aws --type=seldon \
+  --kubernetes_context=zenml-eks --kubernetes_namespace=kubeflow \
+  --base_url=http://$INGRESS_HOST \
+  --secret=s3-store
+zenml container-registry register aws --type=default --uri=715803424590.dkr.ecr.us-east-1.amazonaws.com
```

Comment:
Is this URI meant to be hard-coded?

Reply:
I suspect wdym by hard coded?

Reply:
This URI is hard-coded (`715803424590.dkr.ecr.us-east-1.amazonaws.com`). Actually all of this is hard-coded. There is portion in chapter 2 of zenbytes that makes this completely variable. Check out this for reference: https://github.com/zenml-io/zenbytes/blob/main/02%20-%20Transitioning%20To%20Production.ipynb